examples/dts/dts_activity_sequence.py

```diff
@@ -0,0 +1,73 @@
+import os
+from azure.identity import DefaultAzureCredential
+
+"""End-to-end sample that demonstrates how to configure an orchestrator
+that calls an activity function in a sequence and prints the outputs."""
+from durabletask import client, task, worker
+
+
+def hello(ctx: task.ActivityContext, name: str) -> str:
+    """Activity function that returns a greeting"""
+    return f'Hello {name}!'
+
+
+def sequence(ctx: task.OrchestrationContext, _):
+    """Orchestrator function that calls the 'hello' activity function in a sequence"""
+    # call "hello" activity function in a sequence
+    result1 = yield ctx.call_activity(hello, input='Tokyo')
+    result2 = yield ctx.call_activity(hello, input='Seattle')
+    result3 = yield ctx.call_activity(hello, input='London')
+
+    # return an array of results
+    return [result1, result2, result3]
+
+
+# Read the environment variable
+taskhub_name = os.getenv("TASKHUB")
+
+# Check if the variable exists
+if taskhub_name:
+    print(f"The value of TASKHUB is: {taskhub_name}")
+else:
+    print("TASKHUB is not set. Please set the TASKHUB environment variable to the name of the taskhub you wish to use")
+    print("If you are using windows powershell, run the following: $env:TASKHUB=\"<taskhubname>\"")
+    print("If you are using bash, run the following: export TASKHUB=\"<taskhubname>\"")
+    exit()
+
+# Read the environment variable
+endpoint = os.getenv("ENDPOINT")
+
+# Check if the variable exists
+if endpoint:
+    print(f"The value of ENDPOINT is: {endpoint}")
+else:
+    print("ENDPOINT is not set. Please set the ENDPOINT environment variable to the endpoint of the scheduler")
+    print("If you are using windows powershell, run the following: $env:ENDPOINT=\"<schedulerEndpoint>\"")
+    print("If you are using bash, run the following: export ENDPOINT=\"<schedulerEndpoint>\"")
+    exit()
+
+
+default_credential = DefaultAzureCredential()
+# Define the scope for Azure Resource Manager (ARM)
+arm_scope = "https://durabletask.io/.default"
+
+# Retrieve the access token. Note that this approach doesn't support token refresh and can't be used in production.
+access_token = "Bearer " + default_credential.get_token(arm_scope).token
```

Comment:
The missing refresh token handling makes this sample less practical. Please add an implementation that supports refresh tokens. For instance, a `TokenProvider` that utilizes the [various credential classes](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity?view=azure-python) to ensure the token is always valid by handling expiration and refreshing.

Reply:
I think that is fair - but rather than putting this into the sample (making the sample look unnecessarily complex), why don't we work on a better high level client that wraps all of this seamlessly. The client can take an azure identity TokenCredential as input parameter. If none is provided it can use the DefaultCredential.

My vote would be for doing a follow up PR for a high level client and then updating the DTS samples to use the high level client.